resolversrv/views.py

```
@@ -0,0 +1,470 @@
+        response['service'] = ''
+        response['action'] = 'display'
+        response['links'] = links
+        return self.__return_response(response, 200)
+
+
+    def request_link_type_on_the_fly(self):
+        """
+        for these link types, we only need to format the deterministic link and return it
+
+        :return:
+        """
+        url = self.on_the_fly[self.link_type].format(baseurl=self.baseurl, bibcode=self.bibcode)
+        return self.request_link_type_single_url_toJSON(url)
+
+
+    def request_link_type_article(self, results):
+        """
+        for link type = article, we can have one or many urls
+
+        :param results: result from the query
+        :return:
+        """
+        try:
+            if (len(results) > 0):
+                if (len(results) == 1):
+                    return self.request_link_type_single_url_toJSON(results[0]['url'][0])
+                else:
+                    links = {}
+                    links['count'] = len(results)
+                    links['bibcode'] = self.bibcode
+                    links['link_type'] = self.link_type
+                    records = []
+                    for result in results:
+                        record = {}
+                        record['title'] = result['url'][0]
+                        record['url'] = result['url'][0]
+                        records.append(record)
+                    links['records'] = records
+                    response = {}
+                    # when we have multiple sources of electronic journal, there is no url to log (no service)
+                    response['service'] = ''
+                    response['action'] = 'display'
+                    response['links'] = links
+                    return self.__return_response(response, 200)
+            return self.__return_response({'error': 'did not find any records'}, 404)
+        except Exception as e:
```

try to use the generic 'Exception catching' as little as possible; better to use a specific exception - such as KeyValue, IndexValue - and let the unhandled exception propagate. This means that we'll fail early - it is of course not a hard rule, but as a principle it works well to not swallow exceptions

it is not clear, for example, what exception would you be expecting here

---

fixed